src/main/java/reactor/kafka/receiver/ImmutableReceiverOptions.java

```diff
@@ -940,7 +999,8 @@ public boolean equals(Object object) {
                 && Objects.equals(subscribeTopics, that.subscribeTopics)
                 && Objects.equals(assignTopicPartitions, that.assignTopicPartitions)
                 && Objects.equals(subscribePattern, that.subscribePattern)
-                && Objects.equals(consumerListener, that.consumerListener);
+                && Objects.equals(consumerListener, that.consumerListener)
+                && Objects.equals(pauseAllAfterRebalance, that.pauseAllAfterRebalance);
```

Comment:
```suggestion
                && pauseAllAfterRebalance == that.pauseAllAfterRebalance;
```

Reply:
Also need to change `hashCode()` e.g. `Objects.hash(...) + this.pauseAllAfterRebalance ? 1 : 0`.